### DISCUSSION TITLE
Blazor : Create cascading value using RenderTreeBuilder inside a new RenderFragment

### DISCUSSION THREAD

**Discussion by David-Moreira** [dotnet/aspnetcore#29945] [Q&A]
Hello guys,

I am having difficulties trying to create the following Cascading Value using the Render Tree Builder API, where both the parent and childcontent renderFragments are created dynamically, and where I would like the parent RenderFragment to Cascade itself.

Here's my use case: XTabs component should be a Cascading Value.

            var xTabFragment = new RenderFragment(rBuilder =>
            {
                rBuilder.OpenComponent(0, typeof(XTab));
                rBuilder.AddAttribute(1, nameof(XTab.ChildContent), pageFragment);
                rBuilder.AddAttribute(2, nameof(XTab.Title), xTabTitle);
                rBuilder.CloseComponent();
            });

            _xTabsRenderFragment = new RenderFragment(rBuilder =>
            {
                rBuilder.OpenComponent<XTabs>(0);

                rBuilder.AddAttribute(1, nameof(XTabs.RenderMode), RenderMode);
                rBuilder.AddAttribute(2, nameof(XTabs.CloseTabs), CloseTabs);
                rBuilder.AddAttribute(3, nameof(XTabs.NewTabSetActive), NewTabSetActive);
                rBuilder.AddAttribute(4, nameof(XTabs.ChildContent), xTabFragment);

                rBuilder.AddComponentReferenceCapture(5, compRef => _xTabs = (XTabs)compRef);
                rBuilder.CloseComponent();
            });
I have tried the following snippet inside the XTabsRenderFragment, (with adequate sequence numbers) which I've found on similar threads here, but with no success since the use case on those threads was overriding the BuildRenderTree method instead

```
 rbuilder.OpenComponent<CascadingValue<TValue>>(0);
 rbuilder.AddAttribute(1, "Value", this);
 rbuilder.AddAttribute(2, "ChildContent", (builder2) =>
 {
   builder2.AddContent(3, ChildContent);
 });
 rbuilder.CloseComponent();
```
Can you guys help me figure this out, please?

**Comment by mrlife** [comment]
I guess you will need to create an issue or post somewhere like Stack Overflow.

See: https://github.com/dotnet/aspnetcore/issues/29935

**Reply by David-Moreira** [reply]
Thank you for your answer Mr. Life.
I'll make sure to open an issue. 👍
